scene/gui/line_edit.cpp

```diff
@@ -1738,10 +1738,10 @@ void LineEdit::insert_text_at_caret(String p_text) {
 		int available_chars = max_length - text.length();
 		if (p_text.length() > available_chars) {
 			emit_signal(SNAME("text_change_rejected"), p_text.substr(available_chars));
-			p_text = p_text.substr(0, available_chars);
+			p_text = p_text.left(available_chars);
```

Comment:
`available_chars` is the amount of characters you can add before you hit the character limit. Should be always positive.